mozilla/foundation.mozilla.org
# Pni subcat admin improvements

Closes #7640

Testing:
- head on over to https://foundation.mofostaging.net/cms/snippets/wagtailpages/buyersguideproductcategory/ and observe the seemingly random ordering
- head on over to https://foundation-s-pni-subcat-fruyvu.herokuapp.com/cms/snippets/wagtailpages/buyersguideproductcategory/ and observe that the ordering isn't actually that random.

Also note that several subcategories for toys and games are listed last, grouped on their parent category, but still sorted in-group by their relative sort ordering value.
